### PR TITLE
chore: map routes to tabs

### LINKED ISSUE #62: Header component should be global for all pages

### PR DESCRIPTION
**Pull request by astone123**
Fixes #62 

This PR refactors our routes so that each tab is its own client-side route. This way, users can use the browser navigation buttons and refresh the page and remain where they were. It also decreases the number of DOM elements that need to be loaded on page load.

### CODE REVIEW

File: src/components/FixtureRow.tsx [hunk 1]
```diff
@@ -36,6 +36,8 @@ function FixtureRow({fixture}: FixtureRowProps) {
                 src={fixture.teams.home.logo}
                 alt="Home team logo"
                 style={{maxWidth: '20px'}}
+                width={20}
```
Comment (astone123) [inline src/components/FixtureRow.tsx:39]: Drive-by fix - these should have fixed widths and heights to prevent layout shifts when the images are loading in

File: src/lib/utils.ts [hunk 2]
```diff
@@ -1,13 +1,16 @@
-export function getPreferredTheme() {
+import {Theme} from '../hooks/useTheme';
```
Comment (wweitzel) [inline src/lib/utils.ts:1]: Little bit odd importing from hooks in the /lib directory. Maybe better to make a new file `/lib/theme.ts` and move `getPreferredTheme` function into there along with that `Theme` type.

File: src/index.css [hunk 3]
```diff
@@ -33,3 +33,16 @@ body {
   width: 100%;
   max-width: 800px;
 }
+
+.fade-in {
```
Comment (wweitzel) [inline src/index.css:37]: What's this for? I see where its used in the code but can't tell what it is doing

Comment (astone123) [reply src/index.css:37]: It's making the tab content fade in when we switch routes. We were using a bootstrap class to do this before but it wasn't working with the way that I changed the routing so I had to add it as an animation

**Comment by astone123** [comment]
@wweitzel I spent a while trying to get some animation to work, let me know what you think about this in its current state

**Review by wweitzel — APPROVED** [review]
